File: Python/Lessens4/Task1.py
# LAMDA
# В писке храняться числа. Нужно выбрать только четные числа и составить список пар (число; квадрат числа)
# Например: 123458152338
# Ответ: [(2,4), (8,64), (38,1444)]

# Вариант 1
# list_numbers = [1, 2, 3, 4, 5, 8, 15, 23, 38]
# result = list()
# for i in list_numbers:
#     if i % 2 == 0:
#         result.append((i,i**2))
# print(result)

# ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Вариант2
# def select(f, col):
#     return [f(x) for x in col]

# def where(f, col):
#     return [x for x in col if f(x)]


# list_numbers = [1, 2, 3, 4, 5, 8, 15, 23, 38]
# result = select(int, list_numbers)
# print(result)
# result = where(lambda x: x % 2 == 0, result)
# print(result)
# result = list(select(lambda x: (x, x**2), result))
# print(result)

# ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Вариант3
# def where(f, col):
#     return [x for x in col if f(x)]

# list_numbers = [1, 2, 3, 4, 5, 8, 15, 23, 38]
# result = map(int, list_numbers)

# result = where(lambda x: x % 2 == 0, result)
# print(result)
# result = list(map(lambda x: (x, x**2), result))
# print(result)

# ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Вариант4
list_numbers = [1, 2, 3, 4, 5, 8, 15, 23, 38]
result = map(int, list_numbers)
result = filter(lambda x: x % 2 == 0, result)
result = list(map(lambda x: (x, x**2), result))
print(result)

File: Java/Practice2/Task1.java
package Java.Practice2;

// Дана строка sql-запроса "select * from students where ". Сформируйте часть WHERE этого запроса, 
// используя StringBuilder. Данные для фильтрации приведены ниже в виде json строки.
// Если значение null, то параметр не должен попадать в запрос.
// Параметры для фильтрации: {"name":"Ivanov", "country":"Russia", "city":"Moscow", "age":"null"}

public class Task1 {
    public static void main(String[] args) throws Exception {
        String[] list = lib.ReadLineFromFile("Java\\Practice2\\data.txt");
        System.out.println(list[0]);
        StringBuilder resultSelect = LineInList(list[0]);
        System.out.println(resultSelect);
    }

    public static StringBuilder LineInList(String line) {
        String line1 = line.replace("{", "");
        String line2 = line1.replace("}", "");
        String line3 = line2.replaceAll("\"", "");
        System.out.println(line3);
        StringBuilder result = new StringBuilder("select * from students where ");

        String[] arrayData = line3.split(", ");
        for (int i = 0; i < arrayData.length; i++) {
            String[] arrData = arrayData[i].split(":");
            if (arrData[1].equals("null") == false) {
                if (i != 0) {
                    result.append(", ");
                    result.append(arrData[0]);
                    result.append(" = ");
                    result.append(arrData[1]);
                } else {
                    result.append(arrData[0]);
                    result.append(" = ");
                    result.append(arrData[1]);
                }
            }

        }
        return result;
    }
}